### PR TITLE
Upgrade to oclif core v3

### PR DESCRIPTION
**Pull request by ryandagg**
https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00001cOMR2YAO/view
Had to update the branch protection rules for master to accommodate the new test names.
https://github.com/heroku/heroku-cli-command/settings/branch_protection_rules/1052008
From
test ($OS-matix, lts/gallium) && test ($OS-matix, lts/fermium)
to
test ($OS-matix, 18.x)

### CODE REVIEW

File: src/request-id.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import * as uuid from 'uuid'
 
-export const requestIdHeader = 'Request-Id'
+export const requestIdHeader = 'request-id'
```
Comment (ryandagg) [inline src/request-id.ts:3]: HTTP in Node 18 appears to lower case response headers and does not provide a case-insensitive lookup. Headers are case-insensitive per standards, and this "shouldn't" impact functionality.

File: src/completions.ts [hunk 2]
```diff
@@ -1,11 +1,24 @@
...
+  argv?: string[];
+  config: Interfaces.Config;
+}
+
+export type Completion = {
+  skipCache?: boolean;
+  cacheDuration?: number;
+  cacheKey?(ctx: CompletionContext): Promise<string>;
+  options(ctx: CompletionContext): Promise<string[]>;
+}
```
Comment (ryandagg) [inline src/completions.ts:20]: These interfaces appear to have been completely removed from oclif/core. These were copied from core v2 and work without issue in Heroku CLI.

**Review by k80bowman — APPROVED** [review]

**Review by k80bowman — COMMENTED** [review]
I approved because this looks good, but it looks like you will also need to update the node versions we are running our CI against.

**Review by lancedikson — APPROVED** [review]
Great stuff! I just tested this branch on my new heroku cli plugin, and it works great. The latest `oclif generate` scaffolds a structure for the oclif3-compatible plugins only, and the latest stable heroku-cli/command gave me a types mismatch error in TypeScript. Any chances this PR gets merged any time soon? :) I'd really like to release my plugin, though it would be tricky if one of it's dependencies is pulled from Gh repo and should be built right after installing.

**Comment by eablack** [comment]
Hi @lancedikson I can see you are excited about this (we are too!).. However, there is some other significant work that must be done (primarily with our own plugins) before we can incorporate this into the CLI. The good news is that work has been prioritized and we will be tackling it soon. For now, I'm going to close this PR.

**Pull request closed without merging**